framework/src/DevObj.cpp

```diff
@@ -53,37 +56,88 @@ DevObj::DevObj(const char *name, const char *dev_path, const char *dev_class_pat
 
 	// dev_class_path might be nullptr
 	if (dev_class_path) {
-		m_dev_class_path.assign(dev_class_path);
+		m_dev_class_path = strdup(dev_class_path);
+	}
+	if (name) {
+		m_name = strdup(name);
+	}
+	else {
+		DF_LOG_INFO("ERROR: no name provided to DevObj::DevObj");
+	}
+	if (dev_path) {
+		m_dev_path = strdup(dev_path);
+	}
+	else {
+		DF_LOG_INFO("ERROR: no dev_path provided to DevObj::DevObj");
+	}
+}
+
+DevObj::~DevObj() 
+{
+	m_handle_lock.lock();
+	DFPointerList::Index idx = nullptr;
+	idx = m_handles.next(idx);
+	while (idx != nullptr) {
+		DevHandle *h = reinterpret_cast<DevHandle *>(m_handles.get(idx));
+		if (h->isValid()) {
+			m_handle_lock.unlock();
+			DevMgr::releaseHandle(*h);
+			m_handle_lock.lock();
+		}
+		else {
+			idx = m_handles.erase(idx);
+		}
+		// m_handles may have been modified by DevMgr::releaseHandle()
+		idx = nullptr;
+		idx = m_handles.next(idx);
+	}
+	m_handle_lock.unlock();
+
+	if (isRegistered()) {
+		DevMgr::unregisterDriver(this);
+	}
+
+	if (m_name) {
+		free((void *)m_name);
+	}
+	if (m_dev_path) {
```

Comment:
Result of strdup, so free is a match. I initially couldn't find malloc and drew the wrong conclusions.